### ISSUE TITLE
Failed vagrant-share plugin installation on MacOS due to missing development tools

### ISSUE THREAD

**Issue by psamsotha** [hashicorp/vagrant#12264]
I'm trying to install vagrant-share on MacOS and it's giving me an error saying I need to install development tools. I have no idea what development tools it's referring to.

### Vagrant version

2.2.15

### Host operating system

MacOS 10.14.6

### Debug output

https://gist.github.com/psamsotha/f5f8a05e562a377fa22ac9ee63939c8e

### Expected behavior

Successfully install vagrant-share

### Actual behavior

Fails due to missing "development tools"

"Vagrant failed to install the requested plugin because development tools
are required for installation but are not currently installed on this
machine. Please install development tools and then try this command
again."

### Steps to reproduce

1. Install Vagrant 2.2.15
2. `vagrant plugin install vagrant-share`

### References

None found

**Comment by psamsotha** [comment]
I just caught this line in the debug file

>To see why this extension failed to compile, please check the mkmf.log which can be found here:
>
> /Users/paulsamsotha/.vagrant.d/gems/2.6.6/extensions/x86_64-darwin-19/2.6.0/unf_ext-0.0.7.7/mkmf.log

Here are the contents

```
"clang -o conftest -I/opt/vagrant/embedded/include/ruby-2.6.0/x86_64-darwin19 -I/opt/vagrant/embedded/include/ruby-2.6.0/ruby/backward -I/opt/vagrant/embedded/include/ruby-2.6.0 -I. -I/opt/vagrant/embedded/include -mmacosx-version-min=10.9 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/opt/vagrant/embedded/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -I/opt/vagrant/embedded/include -mmacosx-version-min=10.9 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -fno-common -pipe conftest.c  -L. -L/opt/vagrant/embedded/lib -L/opt/vagrant/embedded/lib -L.  -mmacosx-version-min=10.9 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -fstack-protector-strong -L/opt/vagrant/embedded/lib     -lruby.2.6   "
clang: warning: no such sysroot directory: '/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk' [-Wmissing-sysroot]
In file included from conftest.c:1:
In file included from /opt/vagrant/embedded/include/ruby-2.6.0/ruby.h:33:
In file included from /opt/vagrant/embedded/include/ruby-2.6.0/ruby/ruby.h:29:
/opt/vagrant/embedded/include/ruby-2.6.0/ruby/defines.h:123:10: fatal error: 'stdio.h' file not found
#include <stdio.h>
         ^~~~~~~~~
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */
```

**Comment by psamsotha** [comment]
Found [this](https://github.com/frida/frida/issues/338) and the solution was to run `xcode-select --install` and that seemed to add the missing headers but then I get the following error

```
"clang -o conftest -I/opt/vagrant/embedded/include/ruby-2.6.0/x86_64-darwin19 -I/opt/vagrant/embedded/include/ruby-2.6.0/ruby/backward -I/opt/vagrant/embedded/include/ruby-2.6.0 -I. -I/opt/vagrant/embedded/include -mmacosx-version-min=10.9 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/opt/vagrant/embedded/include -D_XOPEN_SOURCE -D_DARWIN_C_SOURCE -D_DARWIN_UNLIMITED_SELECT -D_REENTRANT   -I/opt/vagrant/embedded/include -mmacosx-version-min=10.9 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -fno-common -pipe conftest.c  -L. -L/opt/vagrant/embedded/lib -L/opt/vagrant/embedded/lib -L.  -mmacosx-version-min=10.9 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -fstack-protector-strong -L/opt/vagrant/embedded/lib     -lruby.2.6   "
ld: malformed file
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib/libSystem.tbd:4:18: error: unknown enumerated scalar
platform:        zippered
                 ^~~~~~~~
 file '/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib/libSystem.tbd'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */
```

and [this solution](https://github.com/ultrajson/ultrajson/issues/332#issuecomment-481910514) does **not** solve it for me.

**Comment by psamsotha** [comment]
So after making the previous changes I ended up deleting the following folder

```
rm -rf /Library/Developer/CommandLineTools
```

Then I installed a different version of the Xcode command line tools. I guess running the commands from previous comments, installed an incompatible version of the command line tools. So I [downloaded and installed 11.3.1 Xcode command line tools](https://medium.com/a-swift-misadventure/how-to-install-multiple-versions-of-xcode-on-the-same-macos-machine-a2836387e57f). Now the installation works fine.
